fix: Relates to #138. Prevents user from importing account that is already in account list

ltfschoen opened the pull request:
* Prevents user from importing account and show the message in the screenshot below when they go to "Import account" and select a JSON Backup Keyfile whose address is already associated with an account that is already in the account list

* Prevents user from importing account and show the message in the screenshot below when they go to "Import account" and enter a Seed Phrase that corresponds to an address that is associated with an account that is already in the account list

<img width="358" alt="screen shot 2018-12-28 at 2 03 42 pm" src="https://user-images.githubusercontent.com/6226175/50501391-ee95f080-0aab-11e9-9683-0779f4969ae8.png">

Tbaut commented on packages/fether-react/src/Accounts/CreateAccount/AccountImportOptions/AccountImportOptions.js:
```diff
@@ -67,6 +90,22 @@ class AccountImportOptions extends Component {
+    if (isExistingAddress) {
+      this.setState({
+        isLoading: false,
+        error: `Account already loaded. Address ${addressForImport} is already in the account list`
```
I would go for something much shorter along the line:
"Account 0x1234..1234 already listed!"
You don't have to see the whole address here.

ltfschoen commented:
@Tbaut Yes, agreed. I personally like using colour-coded animated toasters. Should we create an issue/opportunity for this?
Here are a couple of libraries that include demos:
* https://www.npmjs.com/package/react-toastify
* https://codeseven.github.io/toastr/

ltfschoen commented:
@amaurymartiny @Tbaut I've pushed commits addressing your review comments

amaurymartiny commented on packages/fether-react/src/Accounts/CreateAccount/AccountImportOptions/AccountImportOptions.js:
```diff
@@ -67,6 +90,22 @@ class AccountImportOptions extends Component {
+    const { accounts } = this.props;
+    const isExistingAddress = accounts
+      .map(address => address && address.toLowerCase())
+      .includes(addressForImport);
```
I would put `.toLowecase()` on this line. So that the input argument can be in any case.

amaurymartiny commented on packages/fether-react/src/Accounts/CreateAccount/AccountImportOptions/AccountImportOptions.js:
```diff
@@ -54,7 +69,15 @@ class AccountImportOptions extends Component {
     this.setState({ isLoading: true });
+
     try {
+      const json = JSON.parse(jsonString);
```
Do not parse here.

3 lines below, the `await setJsonString(jsonString);` actually parses the JSON, and createAccountStore.address will hold the parsed address. Do the `hasExistingAddressForImport` then.

amaurymartiny commented:
@ltfschoen  Yes, I agree the errors are quite ugly. In TxForm, I went with react-final-form to put errors as black tooltips over fields, but that might not be ideal for all types of errors. You can create an issue for us to discuss UI options for errors.

Tbaut commented:
@ltfschoen can you make sure to switch the labels to "please review" and remove "grumbles" once you think you've addressed all the comments?

ltfschoen commented:
> @ltfschoen can you make sure to switch the labels to "please review" and remove "grumbles" once you think you've addressed all the comments?

Sure!

ltfschoen commented:
I've merged latest from master

amaurymartiny commented:
@ltfschoen There are 2 grumbles left (from me) 4d ago. I'll merge this PR as soon as they are addressed.
